lib/collection/src/shards/transfer/mod.rs

```diff
@@ -16,6 +27,64 @@ pub trait ShardTransferConsensus: Send + Sync {
     /// Returns `(commit, term)`.
     fn consensus_commit_term(&self) -> (u64, u64);
 
+    /// After snapshot recovery, propose to switch shard to `Partial`
+    ///
+    /// This is called after shard snapshot recovery has been completed on the remote. It submits a
+    /// proposal to consensus to switch the the shard state from `PartialSnapshot` to `Partial`.
+    ///
+    /// # Warning
+    ///
+    /// This only submits a proposal to consensus. Calling this does not guarantee that consensus
+    /// will actually apply the operation across the cluster.
```

Comment:
It may fail to commit the proposal, due to a timeout for example.

Our implementation of a consensus proposal channel does not support getting confirmations back. We just fire and forget hoping it'll eventually be committed. I don't know the internals well enough to know if we can (easily) implement confirmations.